controllers/upgrade_controller.go

```diff
@@ -162,22 +163,19 @@ func (r *UpgradeReconciler) BuildState(ctx context.Context) (*upgrade.ClusterUpg
 		return nil, err
 	}
 
-	for i := range podList.Items {
-		pod := &podList.Items[i]
-		if pod.OwnerReferences == nil || len(pod.OwnerReferences) < 1 {
-			r.Log.V(consts.LogLevelWarning).Info("OFED Driver Pod has no owner DaemonSet", "pod", pod)
-			continue
-		}
-		r.Log.V(consts.LogLevelDebug).Info("Pod", "pod", pod.Name, "owner", pod.OwnerReferences[0].Name)
-
-		ownerDaemonSet, ok := daemonSets[pod.OwnerReferences[0].UID]
-
-		if !ok {
-			r.Log.V(consts.LogLevelWarning).Info("OFED Driver Pod is not owned by an OFED Driver DaemonSet",
-				"pod", pod, "actual owner", pod.OwnerReferences[0])
-			continue
+	filteredPodList := []corev1.Pod{}
+	for _, ds := range daemonSets {
+		dsPods := r.getPodsOwnedbyDs(ds, podList.Items)
+		if int(ds.Status.DesiredNumberScheduled) < len(dsPods) {
+			r.Log.V(consts.LogLevelInfo).Info("Driver daemon set has Unscheduled pods", "name", ds.Name)
+			return nil, fmt.Errorf("DS should not have Unscheduled pods")
```

Comment:
note: this is not really an error but a valid system state we might hit. Ideally we would need to just return Nil and request requeue.